portfolio/src/main/webapp/script.js

```diff
@@ -34,30 +34,57 @@ function gotoRandomSite() {
 }
 
 function getComments() {
-    // Perform the fetch and store as promise
-    var commentsPromise = fetch('/data');
-
-    // Use callbacks on promise to extract the JSON object and put into HTML
-    commentsPromise.then(response => response.json()).then((resJson) => {
-        const commentsContainer = document.getElementById("comments-container");
-        commentsContainer.innerText = '';
-        
-        resJson.forEach(comment => {
-            commentsContainer.appendChild(createParagraphElement(comment));
-        })
+  // Perform the fetch and store as promise
+  var commentsPromise = fetch('/data');
+
+  // Use callbacks on promise to extract the JSON object and put into HTML
+  commentsPromise.then(response => response.json()).then((resJson) => {
+    const commentsContainer = document.getElementById("comments-container");
+    commentsContainer.innerText = '';
+
+    resJson.forEach(comment => {
+      commentsContainer.appendChild(createParagraphElement(comment));
+    })
+  });
+}
+
+function translateComments() {
+  var container = document.getElementById('comments-container');
+  var languageCode = document.getElementById('language').value;
+    const params = new URLSearchParams();
```

Comment:
I'm not sure why you use URLSearchParams here. I think simpler approach (and more natural to read) would be to prepare JSON object and then post it in request body. See "Uploading JSON data" section in https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

Reply:
Sure, I can definitely do that. I just went with what the example code in the walkthrough did, but json does seem easier and more readable.

Reply:
Ummmm I am still trying to work this out, I actually cannot get it to work with using a JSON object and putting it in the request body. On the Java side the getParameter just keeps returning null, but I'm working on it.

Reply:
From what I've gathered, servlets can only get parameters if they are encoded in a certain way of key-value pairs and if I wanted to parse a JSON object I'd have to implement my own decoder after using `request.getReader()` to get a `BufferedReader` that I can iterate over.
[This](https://stackoverflow.com/questions/3831680/httpservletrequest-get-json-post-data) article seems to sum up what I've found over the past couple of hours of bugs lol. Let me know what you think of this situation.

Reply:
Going with URLSearchParams as discussed in the team meeting 👍.

Reply:
Sounds good.